apps/music/js/db.js

```diff
@@ -124,17 +124,49 @@ function initDB() {
       // Hide the  spinner once we've displayed the initial screen
       document.getElementById('spinner-overlay').classList.add('hidden');
 
-      // Concurrently, start scanning for new music
-      musicdb.scan();
-
-      // Only init the communication when music is not in picker mode.
-      if (document.URL.indexOf('#pick') === -1) {
-        // We need to wait to init the music comms until the UI is fully loaded
-        // because the init of music comms could slow down the startup time.
-        MusicComms.init();
+      if (musicdb.state === MediaDB.READY) {
+        startupOnReady();
+      }
+      else {
+        musicdb.onready = startupOnReady;
       }
     });
-  };
+  }
+
+  // This function gets called when we're first launched, after
+  // startupOnEnumerable has run and when or after the mediadb has
+  // reached the ready state.
+  function startupOnReady() {
+    // Start scanning for new music
+    musicdb.scan();
+
+    // Only init the communication when music is not in picker mode.
+    if (document.URL.indexOf('#pick') === -1) {
+      // We need to wait to init the music comms until the UI is fully loaded
+      // because the init of music comms could slow down the startup time.
+      MusicComms.init();
```

Comment:
I'm not sure we can actually wait this long to initialize the comms. We'll usually be safe, but it would be possible to play music before we hit this, so we wouldn't be able to notify the lockscreen that playback has started.